igvc_perception/src/multiclass_segmentation/train.py

```diff
@@ -0,0 +1,219 @@
+    image_arr_path=train_images_path,
+    mask_arr_path=train_masks_path,
+    random_state=420,
+    valid_size=0.1,
+    batch_size=3,
+    num_workers=2,
+)
+
+# Optimize for cross entropy using Adam
+criterion = {
+    "CE": CrossentropyND(),
+}
+
+learning_rate = 0.001
+encoder_learning_rate = 0.0005
+encoder_weight_decay = 0.00003
+optimizer_weight_decay = 0.0003
+optim_factor = 0.25
+optim_patience = 2
+
+optimizer = AdamW(
+    model.parameters(),
+    lr=0.001,
+    betas=(0.9, 0.999),
+    eps=1e-08,
+    weight_decay=0.01,
+    amsgrad=False,
+)
+
+scheduler = optim.lr_scheduler.ReduceLROnPlateau(
+    optimizer, factor=optim_factor, patience=optim_patience
+)
+
+num_epochs = 10
+device = utils.get_device()
+
+runner = SupervisedRunner(device=device, input_key="image", input_target_key="mask")
+
+# Use Catalyst callbacks for metric calculations during training
+callbacks = [
+    CriterionCallback(input_key="mask", prefix="loss", criterion_key="CE"),
+    MulticlassDiceMetricCallback(input_key="mask"),
+]
+
+# Train and print model training logs
+runner.train(
+    model=model,
+    criterion=criterion,
+    optimizer=optimizer,
+    scheduler=scheduler,
+    loaders=loaders,
+    callbacks=callbacks,
+    logdir="content/full_model2",
+    num_epochs=num_epochs,
+    main_metric="loss",
+    minimize_metric=True,
+    fp16=None,
+    monitoring_params=None,
+    verbose=True,
+)
+
+# Test model on test dataset
+test_data = SegmentationDataset(test_images_path, test_masks_path)
+infer_loader = DataLoader(test_data, batch_size=12, shuffle=False, num_workers=4)
+
+# Get model predictions on test dataset
+predictions = np.vstack(
+    list(
+        map(
+            lambda x: x["logits"].cpu().numpy(),
+            runner.predict_loader(
+                loader=infer_loader, resume=f"content/full_model2/checkpoints/best.pth"
+            ),
+        )
+    )
+)
+
+# Display sample prediction results
+pred_results = {}
+rand_nums = np.random.randint(low=1, high=148, size=18)
+rand_nums = np.insert(rand_nums, 0, 30)
+rand_nums = np.insert(rand_nums, 0, 141)
+
+for num in rand_nums:
+    # Show image
+    image = np.asarray(test_data[num]["image"])
+    image = np.swapaxes(image, 2, 0)
+    image = np.swapaxes(image, 1, 0)
+    image = image.astype(np.uint8)
+
+    # Show mask
+    mask = np.squeeze(test_data[num]["mask"])
+
+    # Show predicted mask
+    pred_mask = np.asarray(predictions[num])
+    pred_mask = np.swapaxes(pred_mask, 2, 0)
+    pred_mask = np.swapaxes(pred_mask, 1, 0)
+    pred_mask = pred_mask.astype(np.float64)
+    pred_mask = np.argmax(pred_mask, axis=2)
+
+    # Add three images to list
+    images = []
+    images.append(image)
+    images.append(mask)
+    images.append(pred_mask)
+
+    # Show and plot all three images
+    plt.figure(figsize=(30, 30))
+    columns = 5
+    for i, image in enumerate(images):
+        image_plot = plt.subplot(len(images) / columns + 1, columns, i + 1)
```

Comment:
I'm not too sure what you're trying to do here with this `plt.subplot`

Reply:
I used it to display multiple images in one figure with enough space in between the images.

Reply:
I don't think it's really working rn with the way you have it setup

Reply:
![pred-2](https://user-images.githubusercontent.com/4125463/81146912-39006a80-8f47-11ea-8ab2-1a22ada5310d.png)
right now it looks something like this

Reply:
It's probably Colab-specific; I'll rewrite this

Reply:
I saved the multi-column display for the notebook and simplified the script's implementation. They now use plt.imsave(), which should work outside of Colab.